service/worker/workerdeployment/version_workflow.go

```diff
@@ -219,7 +219,8 @@ func (d *VersionWorkflowRunner) handleUpdateVersionMetadata(ctx workflow.Context
 		d.VersionState.Metadata.Entries = make(map[string]*commonpb.Payload)
 	}
 
-	for key, payload := range args.UpsertEntries {
+	for _, key := range workflow.DeterministicKeys(args.UpsertEntries) {
```

Comment:
I'll bet Go's new iterators allow doing this in one line. This is fine though, we can wait until the sdk adds iterator-based deterministic wrappers.